src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -3826,109 +3693,93 @@ class MavlinkStreamAltitude : public MavlinkStream
 	}
 
 private:
-	MavlinkOrbSubscription *_global_pos_sub;
-	uint64_t _global_pos_time;
-
 	MavlinkOrbSubscription *_local_pos_sub;
-	uint64_t _local_pos_time;
-
 	MavlinkOrbSubscription *_home_sub;
-	uint64_t _home_time;
-
 	MavlinkOrbSubscription *_sensor_sub;
-	uint64_t _sensor_time;
+
+	uint64_t _local_pos_time{0};
+	uint64_t _sensor_time{0};
 
 	/* do not allow top copying this class */
 	MavlinkStreamAltitude(MavlinkStreamAltitude &);
 	MavlinkStreamAltitude &operator = (const MavlinkStreamAltitude &);
 
 protected:
 	explicit MavlinkStreamAltitude(Mavlink *mavlink) : MavlinkStream(mavlink),
-		_global_pos_sub(_mavlink->add_orb_subscription(ORB_ID(vehicle_global_position))),
-		_global_pos_time(0),
 		_local_pos_sub(_mavlink->add_orb_subscription(ORB_ID(vehicle_local_position))),
-		_local_pos_time(0),
 		_home_sub(_mavlink->add_orb_subscription(ORB_ID(home_position))),
-		_home_time(0),
-		_sensor_sub(_mavlink->add_orb_subscription(ORB_ID(sensor_combined))),
-		_sensor_time(0)
+		_sensor_sub(_mavlink->add_orb_subscription(ORB_ID(sensor_combined)))
 	{}
 
 	bool send(const hrt_abstime t)
 	{
 		mavlink_altitude_t msg = {};
-		bool updated = false;
-		float global_alt = 0.0f;
 
-		{
-			struct vehicle_global_position_s global_pos;
-			updated |= _global_pos_sub->update(&_global_pos_time, &global_pos);
+		msg.altitude_monotonic = NAN;
+		msg.altitude_amsl = NAN;
+		msg.altitude_local = NAN;
+		msg.altitude_relative = NAN;
+		msg.altitude_terrain = NAN;
+		msg.bottom_clearance = NAN;
 
-			if (_global_pos_time != 0) {
-				msg.altitude_amsl = global_pos.alt;
-				global_alt = global_pos.alt;
+		// always update monotonic altitude
+		bool sensor_updated = false;
+		sensor_combined_s sensor = {};
+		_sensor_sub->update(&_sensor_time, &sensor);
```

Comment:
`_sensor_time` is not required anymore since we don't check the return value.

Reply:
deleted